featuretools/tests/entityset_tests/test_ww_es.py

```diff
@@ -824,3 +824,30 @@ def test_normalize_ww_init():
 
     assert es['new_df'].ww.name == 'new_df'
     assert es['new_df'].ww.schema.name == 'new_df'
+
+
+def test_update_dataframe_schema():
+    before = pd.DataFrame({
+        'index': [0, 1, 2],
+        'boolean': [True, False, True],
+        'latlong': [(1.0, 2.0), (3.0, 4.0), np.nan],
+    })
+
+    before.ww.init(index='index', logical_types={
+        'boolean': ww.logical_types.Boolean,
+        'latlong': ww.logical_types.LatLong,
+    })
+
+    es = EntitySet()
+    es.add_dataframe(dataframe_name='data', df=before)
+
+    after = pd.DataFrame({
+        'index': [0, 1, 2],
+        'boolean': [0, 1, 0],
+        'latlong': [[5, 6], '7, 8', [np.nan, np.nan]],
+    }).astype({'boolean': int, 'latlong': str})
```

Comment:
I don't think this type casting is actually changing the dtype of the `latlong` column. Seems to be `object`. Maybe we should add another numeric column and use `float64` and `Int64` for before and after just to add another case to confirm it's working as expected? Maybe not strictly necessary but thinking it couldn't hurt.